echopype/commongrid/api.py

```diff
@@ -76,41 +88,102 @@ def compute_MVBS(ds_Sv, range_meter_bin=20, ping_time_bin="20S"):
     ----------
     ds_Sv : xr.Dataset
         dataset containing Sv and ``echo_range`` [m]
-    range_meter_bin : Union[int, float]
-        bin size along ``echo_range`` in meters, default to ``20``
+    range_var: str
+        The variable to use for range binning.
+        Must be one of ``echo_range`` or ``depth``.
+        Note that ``depth`` is only available if the input dataset contains
+        ``depth`` as a data variable.
+    range_bin : str
+        bin size along ``echo_range`` or ``depth`` in meters,
+        default to ``20m``
     ping_time_bin : str
         bin size along ``ping_time``, default to ``20S``
+    method: str
+        The flox strategy for reduction of dask arrays only.
+        See flox `documentation <https://flox.readthedocs.io/en/latest/implementation.html>`_
+        for more details.
+    **kwargs
+        Additional keyword arguments to be passed
+        to flox reduction function.
 
     Returns
     -------
     A dataset containing bin-averaged Sv
     """
 
+    # First check for bin types
+    if not isinstance(range_bin, str):
+        raise TypeError("range_bin must be a string")
+
+    if not isinstance(ping_time_bin, str):
+        raise TypeError("ping_time_bin must be a string")
+
+    # normalize to lower case
+    # for range_bin
+    range_bin = range_bin.strip().lower()
+    # Only matches meters
+    match_obj = re.match(r"(\d+)(\s+)?(m)", range_bin)
+
+    # Do some checks on range meter inputs
+    if match_obj is None:
+        # This shouldn't be other units
+        raise ValueError("Found incompatible units. Must be in meters.")
+
+    # Convert back to float
+    range_bin = float(match_obj.group(1))
```

Comment:
If the numerical part of the argument is a decimal number, this won't work correctly (I tested the regex at https://regex101.com/). It looks like changing the regex to `r"([\d+]*[.,]{0,1}[\d+]*)(\s+)?(m)"` would do the trick. See https://regex101.com/r/asFP2F/1